### PR TITLE
LwM2M: Add support for DTLS + bugfixes

### PR DESCRIPTION
**Pull request by mike-scott**
The following PR establishes the needed changes to support DTLS for the LwM2M library in Zephyr.
As well as fixes a few bugs related to these changes.

### CODE REVIEW

File: ext/lib/crypto/mbedtls/configs/config-coap.h [hunk 1]
```diff
@@ -95,7 +95,11 @@
  * both ends of the connection!  (See comments in "mbedtls/ssl.h".)
  * The optimal size here depends on the typical size of records.
  */
-#define MBEDTLS_SSL_MAX_CONTENT_LEN             256
+#ifndef COAP_MBEDTLS_SSL_MAX_CONTENT_LEN
```
Comment (rbtchc) [inline ext/lib/crypto/mbedtls/configs/config-coap.h]: should be prefixed by **CONFIG_**

File: ext/lib/crypto/mbedtls/configs/config-coap.h [hunk 2]
```diff
@@ -95,7 +95,11 @@
...
+#define MBEDTLS_SSL_MAX_CONTENT_LEN             1500
+#else
+#define MBEDTLS_SSL_MAX_CONTENT_LEN             COAP_MBEDTLS_SSL_MAX_CONTENT_LEN
```
Comment (rbtchc) [inline ext/lib/crypto/mbedtls/configs/config-coap.h]: same here

**Review by rbtchc — CHANGES_REQUESTED** [review]

File: ext/lib/crypto/mbedtls/configs/config-coap.h (hunk 1 above)
Comment (mike-scott) [reply ext/lib/crypto/mbedtls/configs/config-coap.h]: Nice catch.  Fixing.

File: ext/lib/crypto/mbedtls/configs/config-coap.h (hunk 2 above)
Comment (mike-scott) [reply ext/lib/crypto/mbedtls/configs/config-coap.h]: Same.

**Comment by mike-scott** [comment]
V2 changes per @rbtchc review: https://hastebin.com/ipubaconuz.cpp

**Comment by mike-scott** [comment]
Rebased on today's master [no code changes]

**Review by rbtchc — APPROVED** [review]
LGTM

**Comment by mike-scott** [comment]
Perhaps I should split this up into 2 PRs:
- coap config changes
- DTLS support addition for LwM2M

**Comment by mike-scott** [comment]
Rebased on today's master and added "Open Source Foundries" copyright to changed files.  Diff: https://hastebin.com/fejefahine.diff

**Comment by mike-scott** [comment]
Split this into 3 PRs with these 2 being required prior to this working:
https://github.com/zephyrproject-rtos/zephyr/pull/5339
https://github.com/zephyrproject-rtos/zephyr/pull/5340

**Comment by mike-scott** [comment]
Added @dbkinder for doc changes.

File: subsys/net/lib/lwm2m/lwm2m_engine.c [hunk 3]
```diff
@@ -3153,6 +3158,24 @@ void lwm2m_engine_context_init(struct lwm2m_ctx *client_ctx)
...
 
+#if defined(CONFIG_NET_APP_DTLS)
+static int setup_cert(struct net_app_ctx *app_ctx, void *cert)
+{
+#if defined(MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED)
+	struct lwm2m_ctx *client_ctx = CONTAINER_OF(app_ctx,
+						    struct lwm2m_ctx,
+						    net_app_ctx);
+
+	mbedtls_ssl_conf_psk(&app_ctx->tls.mbedtls.conf,
```
Comment (lpereira) [inline subsys/net/lib/lwm2m/lwm2m_engine.c]: This function can fail; shouldn't the error bubble up to net_app?

Comment (mike-scott) [reply subsys/net/lib/lwm2m/lwm2m_engine.c]: Good catch, I'll fix.  This was copy/paste on my part.
FYI: Of the 7 samples that use this, only 2 check it atm.

**Comment by mike-scott** [comment]
Added issue for other samples not checking return value: https://github.com/zephyrproject-rtos/zephyr/issues/5387

**Comment by mike-scott** [comment]
Updated to return error value from mbedtls_ssl_conf_psk() in lwm2m_engine.c
Diff: https://hastebin.com/ipukuwiloj.cpp

File: samples/net/lwm2m_client/README.rst [hunk 4]
```diff
@@ -77,8 +80,32 @@ configurations in ``prj.conf``::
     CONFIG_NET_IPV6=n
     CONFIG_NET_APP_NEED_IPV6=n
 
-Sample output
-=============
+DTLS Support
+============
+
+To build the lwm2m-client sample for QEMU w/ DTLS support do the following:
```
Comment (dbkinder) [inline samples/net/lwm2m_client/README.rst]: no need to abbreviate, so change ``w/`` to ``with``

**Review by dbkinder — CHANGES_REQUESTED** [review]
one doc nit, otherwise looks good.

**Comment by mike-scott** [comment]
Updated w/ doc change per @dbkinder 
Diff: https://hastebin.com/wayocowemu.sql

**Review by lpereira — APPROVED** [review]

**Comment by mike-scott** [comment]
Rebased on today's master.

**Review by nashif — APPROVED** [review]

**Pull request merged**